### PR TITLE
[UsePythonVersion] [UseRubyVersion] prevent to use exact version

### PR DESCRIPTION
**Pull request by sdobrodeev**
**Task name**: UsePythonVersion, UseRubyVersion

**Description**: Added logic for preventing customer to use exact version of Python and Ruby

**Documentation changes required:** (Y) <Please mark if documentation changes are required>

**Added unit tests:** (N) <Please mark if unit tests were added or updated according changes>

**Attached related issue:** (Y) <Please add link to related issue here>

**Checklist**:
- [x] Task version was bumped - please check [instruction](https://github.com/microsoft/azure-pipelines-tasks/tree/master/docs/taskversionbumping.md) how to do it
- [x] Checked that applied changes work as expected

### CODE REVIEW

File: Tasks/UsePythonVersionV0/versionspec.ts [hunk 1]
```diff
@@ -16,4 +18,14 @@ export function desugarDevVersion(versionSpec: string) {
 export function pythonVersionToSemantic(versionSpec: string) {
     const prereleaseVersion = /(\d+\.\d+\.\d+)((?:a|b|rc)\d*)/g;
     return versionSpec.replace(prereleaseVersion, '$1-$2');
+}
+
+/**
+ * Checks if the patch field is present in the version specification
+ *
```
Comment (egor-bryzgalov) [inline Tasks/UsePythonVersionV0/versionspec.ts:25]: Nit: redundant line

File: Tasks/UseRubyVersionV0/userubyversion.ts [hunk 2]
```diff
@@ -11,6 +12,16 @@ export enum Platform {
     Linux
 }
 
+/**
+ * Checks if the patch field is present in the version specification
+ *
```
Comment (egor-bryzgalov) [inline Tasks/UseRubyVersionV0/userubyversion.ts:17]: Nit: redundant line

**Review by egor-bryzgalov — APPROVED** [review]
LGTM, thank you!

**Review by anatolybolshakov — APPROVED** [review]

File: Tasks/UsePythonVersionV0/Strings/resources.resjson/en-US/resources.resjson [hunk 3]
```diff
@@ -16,5 +16,6 @@
   "loc.messages.PyPyNotFound": "Could not find an installation of PyPy%d in Agent.ToolsDirectory.",
   "loc.messages.ToolNotFoundMicrosoftHosted": "If this is a Microsoft-hosted agent, check that this image supports side-by-side versions of %s at %s.",
   "loc.messages.ToolNotFoundSelfHosted": "If this is a self-hosted agent, see how to configure side-by-side %s versions at %s.",
-  "loc.messages.VersionNotFound": "Version spec %s for architecture %s did not match any version in Agent.ToolsDirectory."
+  "loc.messages.VersionNotFound": "Version spec %s for architecture %s did not match any version in Agent.ToolsDirectory.",
+  "loc.messages.ExactVersionNotRecommended": "It is not recommended to provide an exact version specification. Please try to use the version specification without the patch field"
```
Comment (maxim-lobanov) [inline Tasks/UsePythonVersionV0/Strings/resources.resjson/en-US/resources.resjson:20]: I suggest to rephrase error message a bit:
```
It is not recommended to specify exact version on Microsoft-Hosted agents. Patch version of Python can be replaced by new one on Hosted agents without notice and build stops to work. it is recommended to specify only major or major and minor version (Example: `3` or `3.6`)
```

Comment (maxim-lobanov) [reply Tasks/UsePythonVersionV0/Strings/resources.resjson/en-US/resources.resjson:20]: The same for Ruby message

**Review by maxim-lobanov — APPROVED** [review]

**Pull request merged**